### PR TITLE
[Merged by Bors] - feat: drop an unneeded field of `StarOrderedRing`

### PR DESCRIPTION
**Pull request by urkud**
In `StarOrderedRing.ofLeIff`, the assumption
`h_add : ∀ {x y}, x ≤ y → ∀ z, z + x ≤ z + y` follows from the other one.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Star/Order.lean
Comment (j-loreaux) [inline Mathlib/Algebra/Star/Order.lean:90]:
```suggestion
Porting note: dropped an unneeded assumption `h_add : ∀ {x y}, x ≤ y → ∀ z, z + x ≤ z + y`
```

**Review by j-loreaux — APPROVED** [review]
Great! Thanks for removing this hypothesis.

~~maintainer merge~~

I have edited this PR now and it needs another reviewer. Sorry!

**Comment by j-loreaux** [comment]
actually, please wait on this, I realized that this means `StarOrderedRing` doesn't even need one of its fields.

**Comment by dupuisf** [comment]
Very nice, thanks!

bors r+

**Pull request closed without merging**
